test/srt/models/test_molmo.py

```diff
@@ -0,0 +1,129 @@
+        for name, param in model.model.vision_backbone.named_parameters():
+            if "image_projector" in name:
+                if "w1" in name:
+                    image_projector_weights["gate_proj"] = param
+                elif "w3" in name:
+                    image_projector_weights["up_proj"] = param
+                elif "w2" in name:
+                    image_projector_weights["down_proj"] = param
+                else:
+                    raise ValueError(f"Unexpected projector weight: {name}")
+            else:
+                vision_backbone_weights[name] = param
+
+        # Load weights into our custom MolmoVisionBackbone
+        molmo_vision_backbone.load_state_dict(vision_backbone_weights, strict=False)
+        molmo_vision_backbone.to(device="cuda", dtype=torch.float32)
+
+        for k, v in molmo_vision_backbone.state_dict().items():
+            if k in model.model.vision_backbone.state_dict():
+                assert torch.allclose(
+                    v, model.model.vision_backbone.state_dict()[k], atol=1e-5
+                ), f"Weight {k} is not close"
+            else:
+                continue
+
+        # Load image projector weights
+        gate_up_proj_weight = torch.cat(
+            [image_projector_weights["gate_proj"], image_projector_weights["up_proj"]],
+            dim=0,
+        )
+        molmo_vision_backbone.image_projector.gate_up_proj.weight.data.copy_(
+            gate_up_proj_weight
+        )
+        molmo_vision_backbone.image_projector.down_proj.weight.data.copy_(
+            image_projector_weights["down_proj"]
+        )
+
+        model.model.vision_backbone.eval()
+        molmo_vision_backbone.eval()
+
+        # Test forward pass
+        inputs = processor.process(images=images, text=text, return_tensors="pt")
+        with torch.no_grad():
+            images = inputs["images"].unsqueeze(0).to("cuda")
+            image_masks = inputs["image_masks"].unsqueeze(0).to("cuda")
+
+            hf_outputs_encode_image = model.model.vision_backbone.encode_image(
+                images=images
+            )
+            srt_outputs_encode_image = molmo_vision_backbone.encode_image(images=images)
+
+            hf_outputs = model.model.vision_backbone(
+                images=images, image_masks=image_masks
+            )
+            srt_outputs = molmo_vision_backbone(images=images, image_masks=image_masks)
+
+            # Compare outputs
+            assert (
+                torch.max(abs(hf_outputs[0] - srt_outputs[0])) < 1e-3
+            ), "image_features are not all close"
+            assert (
+                torch.max(abs(hf_outputs[1] - srt_outputs[1])) < 1e-3
+            ), "cls_embed are not all close"
+
+    def test_molmo_d(self):
+        for text, images in CONVS:
+            for model_path, torch_dtype, tolerance in MODELS:
+                self.assert_srt_vision_backbone_and_hf_vision_backbone_close(
```

Comment:
Do you have an end-to-end test for the whole model?
Can you add a new test here https://github.com/sgl-project/sglang/blob/main/test/srt/test_vision_openai_server.py?